File: DSA-II/Lab1 - Intro Python/q1.py
def fact_iter(n):
	fact = 1
	for i in range(1,n+1):
		fact*=i
	return fact

def fact_rec(n):
 	while(n>0):
	 	if (n <= 1):
	 		return 1
	 	return n*fact_rec(n-1)

def main():
	n = int(input())
	print("Iterative: " + str(fact_iter(n)))
	print("Recursive: " + str(fact_rec(n)))

main()

File: DSA-I/Lab1-Revision/q1.cpp
#include<bits/stdc++.h>
using namespace std;

int factorial(int x)
{
	if(x==0 || x==1)
		return 1;
	return x*factorial(x-1);
}
int main()
{
	int n,fact=1,y;
	cout<<"Enter a no"<<endl;
	cin>>n;
	y=n;
	while(n>0)
	{
		fact*=n;
		n--;
	}
	cout<<"Factorial(iterative): "<<fact<<endl;
	cout<<"Factorial(recursive): "<<factorial(y)<<endl;
}